### PR TITLE
Add assetInfo service

### PR DESCRIPTION
**Pull request by JIOjosBG**
Service for token/nft info batch fetching

can be tested with this for benzin
https://github.com/AmbireTech/ambire-app/pull/2548

### CODE REVIEW

File: src/services/assetInfo/assetInfo.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { GetOptions, Portfolio } from '../../libs/portfolio'
+
+const scheduledActions: {
+  [network in NetworkId]?: {
+    promise: Promise<any>
+    data: { callback: Function; address: string }[]
+  }
+} = {}
+
+export async function executeBatchedFetch(network: Network): Promise<any> {
```
Comment (PetromirDev) [inline src/services/assetInfo/assetInfo.ts:13]: Replace any with a strict type. This applies for both PRs

File: src/services/assetInfo/assetInfo.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const UNISWAP_ROUTER = '0xE592427A0AEce92De3Edee1F18E0157C05861564'
+
+global.fetch = fetch as any
+
+describe('Asset info service', () => {
+  test.only('Fetches all tokens and NFTS correctly', async () => {
+    jest.spyOn(assetInfo, 'executeBatchedFetch')
+
+    const wethCallback = jest.fn(({ tokenInfo }) =>
+      expect(tokenInfo).toMatchObject({ symbol: 'WETH', decimals: 18 })
```
Comment (PetromirDev) [inline src/services/assetInfo/assetInfo.test.ts:20]: Write a batch test. Similar to the one written for portfolio lib

File: src/libs/portfolio/portfolio.ts [hunk 3]
```diff
@@ -306,7 +306,7 @@ export class Portfolio {
       tokenErrors: tokensWithErrResult
         .filter(([error, result]: [string, TokenResult]) => error !== '0x' || result.symbol === '')
         .map(([error, result]: [string, TokenResult]) => ({ error, address: result.address })),
-      collections: collections.filter((x) => x.collectibles?.length)
+      collections
```
Comment (PetromirDev) [inline src/libs/portfolio/portfolio.ts:309]: Why?

File: src/services/assetInfo/assetInfo.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    disableAutoDiscovery: true,
+    previousHints: {
+      erc20s: allAddresses,
+      erc721s: Object.fromEntries(
+        allAddresses.map((i) => [
+          i,
+          {
+            tokens: ['1'],
+            isKnown: false
+          }
```
Comment (PetromirDev) [inline src/services/assetInfo/assetInfo.ts:28]: Can you explain this part?

**Review by PetromirDev — CHANGES_REQUESTED** [review]

File: src/libs/portfolio/portfolio.ts (hunk 3 above)
Comment (JIOjosBG) [reply src/libs/portfolio/portfolio.ts:309]: So we get collection info regardless if there is any changes to nft ownership. 
This check is moved to the dashboard component so as we need it only there

File: src/services/assetInfo/assetInfo.ts (hunk 4 above)
Comment (JIOjosBG) [reply src/services/assetInfo/assetInfo.ts:28]: sentinel value, used for single nft fetch from the portfolio

**Review by PetromirDev — APPROVED** [review]

**Pull request merged**